Prod 2450/how to use plural docs

seemywingz opened the pull request:
<!--- Hello Plural contributor! It's great to have you on board! -->
## Summary
Add `How To` for _`plural up`_ , adding a workload cluster, scm connection, controllers and a few placeholders
<!-- Describe your changes here, include the motivation/context, test coverage, -->
<!-- the type of change i.e. breaking change, new feature, or bug fix -->
<!-- and related GitHub issue or screenshots (if applicable). -->

<!-- Adding a meaningful title and description allows us to better communicate -->
<!-- your work with our users. -->

## Labels
enhancement
<!-- For breaking changes, add the `breaking-change` label.️ -->
<!-- For bug fixes, add the `bug-fix` label. -->
<!-- For new features and notable changes, add the `enhancement` label. -->


## Test Plan
<!--- Please describe the tests you have added and your testing environment (if applicable). -->
Viwed changes locally with `yarn dev`

## Checklist
<!--- Go over all the following points to make sure you've checked all that apply before merging. -->
<!--- If you're unsure about any of these, don't hesitate to ask in our Discord. -->

- [x] If required, I have updated the Plural documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] I have added a meaningful title and summary to convey the impact of this PR to a user.
- [x I have added relevant labels to this PR to help with categorization for release notes.

michaeljguarino commented on pages/how-to/set-up/new-mgmt-cluster.md:
````diff
@@ -0,0 +1,104 @@
+```
+
+
+### Create a New Repo for Plural
````
We shouldn't recommend they do this, instead let `plural up` create the repo for them, which it does via oauth (and explain it can be cloned first if they can't do that due to permissioning constraints).

michaeljguarino commented on pages/how-to/set-up/new-mgmt-cluster.md:
````diff
@@ -0,0 +1,104 @@
+### Get Kubeconfig for the MGMT Cluster
+AWS 
+```sh
+aws eks update-kubeconfig --name [CLUSTER_NAME] 
````
`plural wkspace kube-init` does that for you actually

seemywingz replied:
this is really cool!
I didn't know this feature existed

michaeljguarino commented on pages/how-to/set-up/new-mgmt-cluster.md:
````diff
@@ -0,0 +1,104 @@
+
+##### Add the RBAC Helm Values in the MGMT Cluster Repo
+```sh
+./helm-values/ConsoleClusterRoleBindings.yaml
````
do *not* put this in the `helm-values` folder, that's git-crypted by default so should only be used for secret-ish things, create a new global service in a separate folder for rbac which contains this.  Also don't camel-case filenames for k8s resources (not standard convention)

seemywingz replied:
I reference that Here: https://github.com/pluralsh/documentation/pull/289/files/3356e70ddd2f2e93b0baf37b6b500b45b16bcba6#diff-eb5fe659cd5bbca6c330ae4d4e9afe8380837b105bef0a1b1f8c69d126661e7dR97

should the doc just link to that or reiterate what the RBAC readme says?

seemywingz replied:
I think this looks and reads better:
![image](https://github.com/user-attachments/assets/4e671189-a573-4f4d-bda6-00bdd6b9f607)

seemywingz replied:
and links to: https://github.com/pluralsh/documentation/blob/8e205adfede17b0e412a2c8d81ac511dd71fe59b/pages/deployments/dashboard.md

seemywingz commented:
I added the docs for `How To/Set Up` a New [MGMT Cluster](https://github.com/pluralsh/documentation/blob/PROD-2450/how-to-use-plural-docs/pages/how-to/set-up/new-mgmt-cluster.md), I also added installing the Plural CLI and Console. there are also placeholders for `How To/Use` which I think are useful

the placeholders are no added in the NavData

seemywingz commented:
@michaeljguarino I made the requested changes

michaeljguarino commented on src/NavData.tsx:
```diff
@@ -100,6 +100,64 @@ const rootNavData: NavMenu = deepFreeze([
+                href: '/how-to/set-up/plural-cli',
+              },
+              {
+                title: 'MGMT Cluster',
```
These titles need some real work, the terms are not going to be understood by novice users and it'll be the first thing they see when browsing the docs

seemywingz replied:
addressed this by changing the nav name and in the docs wrote `Management (MGMT)` so the reference is clear

michaeljguarino commented on pages/how-to/set-up/mgmt-cluster.md:
```diff
@@ -0,0 +1,136 @@
+### Prerequisites
+[Plural CLI](/how-to/set-up/plural-cli)
+
+##### Ensure Cloud Provider CLI Authentication
```
I believe we have some existing docs on how to set up cloud cli authentication as well we should link to or just duplicate, that's a bit better formatted as well

seemywingz replied:
linked to the existing cli install

michaeljguarino commented on pages/how-to/set-up/mgmt-cluster.md:
```diff
@@ -0,0 +1,136 @@
+
+### "Cannot list resources in the Kubernetes Dashboard"
+![alt text](/images/how-to/k8s-dash-403.png)
+This is expected and due to missing [RBAC Bindings](https://github.com/pluralsh/documentation/blob/main/pages/deployments/dashboard.md) for the console users  
```
use relative link (this is available in the doc site)

michaeljguarino replied:
Also seems to be just reanswered in the section below

seemywingz replied:
addressed the url.
This was intended as a header or new section block for the steps below

michaeljguarino commented on pages/how-to/set-up/mgmt-cluster.md:
```diff
@@ -0,0 +1,136 @@
+
+##### Creating an RBAC Service
+* **Create an `rbac` dir in your MGMT repo 
+and add the desired [k8s yaml](https://github.com/pluralsh/documentation/blob/main/pages/deployments/dashboard.md)** 
```
relative link

seemywingz replied:
address

michaeljguarino commented on pages/how-to/set-up/mgmt-cluster.md:
```diff
@@ -0,0 +1,136 @@
+  name: cluster-admin
+subjects:
+  - apiGroup: rbac.authorization.k8s.io
+    kind: User
```
use group bindnigs too to explain that

michaeljguarino commented on pages/how-to/set-up/pipelines.md:
```diff
@@ -0,0 +1,50 @@
+# Prerequisites
+[Plural CLI](/how-to/set-up/plural-cli)
+
+#### Ensure Cloud Provider CLI Authentication
```
shouldn't need to duplicate this here

michaeljguarino commented on pages/how-to/set-up/plural-cli.md:
```diff
@@ -0,0 +1,28 @@
+description: Guides for installing the Plural CLI
+---
+
+### Install Prerequisites
```
use existing docs (they have nicer formatting)

michaeljguarino commented on pages/how-to/set-up/scm-connection.md:
```diff
@@ -0,0 +1,91 @@
+  * [Bitbucket](https://support.atlassian.com/bitbucket-cloud/docs/access-tokens/)
+* **SCM Provider Organization `admin` Permissions**
+  * This is only required when creating the webhook
+  * The workload cluster can still be created without the SCM webhook
```
This is jumping the gun (we aren't in the workload cluster docs).  Just say the SCM connection can still function without a webhook, you just won't receive PR merge/close updates from the provider.

michaeljguarino commented on pages/how-to/set-up/scm-connection.md:
```diff
@@ -0,0 +1,91 @@
+* **Fil in the Required Fields**
+  * **Provider Type**: The SCM Provider Hosting Git Repositories  
+  * **Name**: Reference Name for the Provider  
+    * ℹ️ **NOTE**: The _cluster-creator_ PR Automation looks for `github` by default
```
(say you can change that name if you want to match your scm provider)

michaeljguarino commented on pages/how-to/set-up/scm-connection.md:
```diff
@@ -0,0 +1,91 @@
+### **Create an [`ScmConnection`](https://docs.plural.sh/deployments/operator/api#scmconnection) CRD Instance**  
+Once the connection is created in the UI we can reference it with a CRD instance
+  * ❕ Ensure the Name Provided in the UI matches the `spec.name` in the CRD Exactly
+  * An [`ScmConnection`](https://docs.plural.sh/deployments/operator/api#scmconnection) yaml template for GitHub exists in `./app/services/pr-automation/scm.yaml`  
```
relative link

michaeljguarino commented on pages/how-to/set-up/scm-connection.md:
````diff
@@ -0,0 +1,91 @@
+  type: GITHUB
+```
+### **Add an SCM Provider Webhook**
+If you navigate to `https://console.[YOUR DOMAIN].onplural.sh/pr/queue`  
````
make these link urls: `https://{your-console-domain}/pr/queue` (most serious users bring their own domain)

michaeljguarino commented on pages/how-to/set-up/scm-connection.md:
```diff
@@ -0,0 +1,91 @@
+
+# Troubleshooting
+
+#### Get Kubeconfig for the MGMT Cluster
```
don't make them troubleshoot w/ kubectl, they should be able to see this info in the CD ui (can also add the kubectl command as an addendum)

michaeljguarino commented on pages/how-to/set-up/workload-cluster.md:
```diff
@@ -0,0 +1,53 @@
+
+
+
+# Troubleshooting
```
incomplete

michaeljguarino commented on pages/how-to/set-up/workload-cluster.md:
```diff
@@ -0,0 +1,53 @@
+![cluster-creator modal 2](/images/how-to/cluster-creator-modal-2.png)
+* **Merge the PR**
+* **Approve the Stack Run changes**
+  * Navigate to `https://console.[YOUR DOMAIN].onplural.sh/staacks`
```
use console urls formatted as above

michaeljguarino commented on pages/how-to/upgrade/plural-cli.md:
```diff
@@ -0,0 +1,28 @@
+---
+title: Upgrading the Plural CLI
```
don't think we need these docs or at least not here if it's going to rabbithole users

michaeljguarino reviewed (commented):
High level comments:

* honestly feel like the docs for setting up a management cluster should probably just link to our already thorough docs elsewhere
* there's a lot of incomplete content here I don't think we can merge and expose publicly

michaeljguarino commented on pages/how-to/set-up/controllers.md:
````diff
@@ -0,0 +1,119 @@
+* **Add a new `HelmRepository` CRD to your _infra_ repo `./apps/repositories`**
+  * Example: `ingress-nginx.yaml`
+```yaml
+apiVersion: source.toolkit.fluxcd.io/v1beta2
````
This should use the native helm support not flux-based now that that's fully released.  So this resource isn't necessary, and the `repository` field in the helm block below can just be replaced with`url` pointing to the url in this CRD

michaeljguarino commented on src/NavData.tsx:
```diff
@@ -100,6 +100,39 @@ const rootNavData: NavMenu = deepFreeze([
+                href: '/how-to/set-up/workload-cluster',
+              },
+              {
+                title: 'SCM Connection',
```
you'll need to set up the scm connection before the workload cluster since it's a prereq.  Also many users aren't going to understand the acronym, this should be titled something like `Source Control Integration` or something like that.

michaeljguarino commented on pages/how-to/set-up/scm-connection.md:
```diff
@@ -0,0 +1,68 @@
+# Prerequisites
+* **Plural Console `admin` Permissions**  
+* **SCM Provider Personal Access Token**
+  * [Github](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens#creating-a-personal-access-token-classic)  
```
For github we should strongly nudge people to configure it via a github application which is technically more secure and not bound to a specific user.

michaeljguarino commented on pages/how-to/set-up/scm-connection.md:
```diff
@@ -0,0 +1,68 @@
+* **Plural Console `admin` Permissions**  
+* **SCM Provider Personal Access Token**
+  * [Github](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens#creating-a-personal-access-token-classic)  
+  * [GitLab](https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html#create-a-personal-access-token)  
```
This should use group or project tokens not personal tokens

michaeljguarino commented on pages/how-to/set-up/scm-connection.md:
```diff
@@ -0,0 +1,68 @@
+* **SCM Provider Personal Access Token**
+  * [Github](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens#creating-a-personal-access-token-classic)  
+  * [GitLab](https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html#create-a-personal-access-token)  
+  * [Bitbucket](https://support.atlassian.com/bitbucket-cloud/docs/access-tokens/)
```
same, bitbucket has repo and org tokens that can be used
